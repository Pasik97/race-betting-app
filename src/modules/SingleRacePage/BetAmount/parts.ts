import styled from "styled-components";

export const InputWrapper = styled.div`
   display: flex;
   margin: 24px 0;
`;

export const Label = styled.p`
   margin: 0 8px 0 0;
   font-size: 16px;
   line-height: 18px;
   color: #1a1a1a;
`;

export const Input = styled.input`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   padding: 0;
   border: none;
   outline: none;
   border-bottom: 1px solid #dddddd;

   &:focus {
      border-bottom: 1px solid #1a1a1a;
   }
`;

