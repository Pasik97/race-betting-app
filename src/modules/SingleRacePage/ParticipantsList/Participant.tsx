import React from 'react';

const Participant: React.FC = () => {
   return null;
}

export default Participant;
