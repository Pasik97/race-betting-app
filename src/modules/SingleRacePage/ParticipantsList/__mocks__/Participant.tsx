const mock = () => '(Participant /)';

export default mock;
