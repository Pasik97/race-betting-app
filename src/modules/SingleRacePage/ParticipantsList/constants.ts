export interface ParticipantsListProps {
   raceId: string;
};

export interface ParticipantProps {
   participantId: number;
   raceId: string;
}

export const placesEmojis = ['ðŸ¥‡', 'ðŸ¥ˆ', 'ðŸ¥‰'];
