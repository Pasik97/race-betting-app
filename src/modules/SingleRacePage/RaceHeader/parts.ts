import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const RaceTitleWrapper = styled.div`
   display: flex;
   align-items: baseline;
   justify-content: space-between;
`;

export const RaceTitle = styled.h2`
   color: #01263A;
   font-weight: 400;
   font-size: 24px;
   line-height: 27px;
   margin: 16px 0 0 0;
`;

export const RaceTitleStatus = styled.p`
   margin: 0 0 0 8px;
   font-size: 16px; 
   line-height: 18px;
   white-space: nowrap;
`;

export const StyledLink = styled(Link)`
   display: inline-flex;
   align-items: center;
   height: 32px;
   color: #1a1a1a;
   font-size: 13px;
   line-height: 16px;
   text-decoration: none;
   background: #f5f5f5;
   padding: 0px 20px;
   border-radius: 24px;

   &:hover {
      background: #eeeeee;
   }
`;
