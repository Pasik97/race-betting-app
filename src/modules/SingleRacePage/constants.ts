import { RouteComponentProps } from "react-router-dom";

interface SingleRacePageParams {
   id: string;
}

export type SingleRacePageProps = RouteComponentProps<SingleRacePageParams>;
