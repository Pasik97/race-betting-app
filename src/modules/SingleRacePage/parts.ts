import styled from "styled-components";

export const SingleRacePageWrapper = styled.div`
    position: relative;
    min-height: 150px;
`;
