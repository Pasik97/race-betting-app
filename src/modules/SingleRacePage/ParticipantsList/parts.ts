import styled from 'styled-components';
import { Row } from 'components/BasicTable/parts';
import media from 'utils/media';

export const ParticipantsListWrapper = styled.div``;

export const StyledRow = styled(Row)`
   grid-template-columns: 0.5fr 2fr 0.5fr 0.5fr 0.5fr;

   ${media.fromTablet} {
      grid-template-columns: 0.5fr 4fr 0.5fr 0.5fr 0.5fr;
   }

   ${media.fromDesktop} {
      grid-template-columns: 0.5fr 6fr 0.5fr 0.5fr 0.5fr;
   }
`;

export const Radio = styled.input``;
