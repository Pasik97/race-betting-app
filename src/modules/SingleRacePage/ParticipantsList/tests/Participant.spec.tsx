import { act, create } from "react-test-renderer";
import { Provider } from "react-redux";
import Participant from "../Participant";
import { Radio } from "../parts";
import * as A from 'store/races/actions';
import { createMockRaceStore } from "modules/SingleRacePage/tests/mockData";
import { mockRace, mockRacesState } from "store/races/tests/mockData";
import { mockParticipant, mockParticipantsState } from "store/participants/tests/mockData";
import { Places } from "store/races/constants";

describe('<Participant />', () => {
   it('should render participant id name and 3 inputs with type set to radio', () => {
      const component = create(
         <Provider store={createMockRaceStore(mockRacesState, mockParticipantsState)}>
            <Participant raceId={mockRace.id.toString()} participantId={mockParticipant.id} />
         </Provider>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });

   describe('onRadioChange', () => {
      it('should dispatch setRaceBetPlace', () => {
         const setRaceBetPlaceSpy = jest.spyOn(A, 'setRaceBetPlace');

         const component = create(
            <Provider store={createMockRaceStore(mockRacesState, mockParticipantsState)}>
            <Participant raceId={mockRace.id.toString()} participantId={mockParticipant.id} />
         </Provider>,
         );
         const radio = component.root.findAllByType(Radio)[0];
         
         act(() => {
            radio.props.onChange(Places.First);
         });

         expect(setRaceBetPlaceSpy).toHaveBeenCalled();
      });
   });
});
