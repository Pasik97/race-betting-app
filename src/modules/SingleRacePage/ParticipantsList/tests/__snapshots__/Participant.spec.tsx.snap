// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participant /> should render participant id name and 3 inputs with type set to radio 1`] = `
<div
  className="sc-dkPtRN sc-jRQBWg bmFaAY dgjLOM"
>
  <div
    className="sc-bdvvtL YGxmD"
  >
    1
  </div>
  <div
    className="sc-bdvvtL YGxmD"
  >
    mock body
  </div>
  <div
    className="sc-bdvvtL YGxmD"
  >
    <input
      checked={false}
      className="sc-gKclnd"
      disabled={false}
      onChange={[Function]}
      type="radio"
    />
  </div>
  <div
    className="sc-bdvvtL YGxmD"
  >
    <input
      checked={false}
      className="sc-gKclnd"
      disabled={false}
      onChange={[Function]}
      type="radio"
    />
  </div>
  <div
    className="sc-bdvvtL YGxmD"
  >
    <input
      checked={false}
      className="sc-gKclnd"
      disabled={false}
      onChange={[Function]}
      type="radio"
    />
  </div>
</div>
`;
